{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "### info from oxyphen developer:\n",
    "\n",
    "#To run multinome version, you just have to run the script oxyphen_mutlinome.py \n",
    "#and in SETTINGS file add a line (“PROTEOMES_FOLDER”) that points to the folder with your proteomes in fasta format.\n",
    "#SETTINGS:\n",
    "#INPUT_FILE=/Users/jagodajablonska/Desktop/oxygen_paper/oxyphen/PROTEOMES/UP000245539_1247513.fasta\n",
    "#PROTEOMES_FOLDER=/Users/jagodajablonska/Desktop/oxygen_paper/oxyphen/PROTEOMES\n",
    "#BLAST_PATH=/usr/local/ncbi/blast/bin\n",
    "#NUM_THREADS=2\n",
    "\n",
    "#The program will save separate files with output oxygen-utilizing EC classes in OUTPUT directory \n",
    "#(and detailed mapping) for every input proteome and display the number of them on the screen. \n",
    "\n",
    "#I want to avoid returning explicitly whether the organism is aerobic or not.  \n",
    "#I’d stick to the publication data and assume that an organism with more than around 25 oxygen-utillizing enzymes \n",
    "#is with high confidence an aerobe.\n",
    "\n",
    "#and here's the BLAST executables handbook. Python script below is running command line blast commands from downloaded blast exe\n",
    "# http://nebc.nerc.ac.uk/bioinformatics/documentation/blast+/user_manual.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# notes to self:\n",
    "#before running this script, complete four steps:\n",
    "\n",
    "#Use API scrape (separate jupyter notebook) to get proteome csv from uniprot website\n",
    "#use for loop script (R) to convert csv files to fasta files.\n",
    "#rename / discard old folders and files, and remake new OUTPUT folder.\n",
    "#edit line 122-123 below to set protein homology threshold for greater specificity (high, 60) or sensitivity (low, 40).\n",
    "\n",
    "#after running this oxyphen multiome script, run another R script to merge results summary table with org phylogeny files\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/tonglen/Downloads/oxyphen-multiome'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "path='/Users/tonglen/downloads/oxyphen-multiome/'\n",
    "os.chdir(path)\n",
    "\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from sklearn import svm\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import classification_report, confusion_matrix\n",
    "from sklearn.feature_selection import RFE\n",
    "from Bio.ExPASy import Enzyme\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os, glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1076588.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "8 oxygen-utilizing enzymes were found from classes dict_keys(['1.14.13.7', '1.15.1.1', '4.1.1.39', '1.4.3.16', '1.14.13.236', '1.9.3.1', '1.3.3.3', '1.10.2.2'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1076588_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1076588_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1163617.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "9 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21', '1.4.3.16', '4.1.1.39', '1.11.1.6', '1.15.1.1', '1.9.3.1', '1.7.2.1', '1.3.3.3', '1.10.2.2'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1163617_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1163617_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1223802.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "10 oxygen-utilizing enzymes were found from classes dict_keys(['1.7.2.1', '1.9.3.1', '1.13.11.8', '1.14.13.208', '1.4.3.16', '4.1.1.39', '1.3.3.3', '1.10.2.2', '1.14.13.2', '1.15.1.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1223802_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1223802_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1282737.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "3 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.9.3.1', '1.15.1.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1282737_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1282737_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1303921.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "4 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.4.3.5', '1.3.3.3', '1.10.2.2'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1303921_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1303921_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1434104.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "1 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1434104_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1434104_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1564506.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "11 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.16.3.1', '1.15.1.1', '1.3.99.28', '1.4.3.5', '1.10.2.2', '1.9.3.1', '1.11.1.21', '1.3.3.11', '1.10.3.14', '1.5.3.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1564506_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1564506_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/159087.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "15 oxygen-utilizing enzymes were found from classes dict_keys(['1.13.11.49', '1.14.13.236', '1.3.3.11', '1.14.13.7', '1.9.3.1', '1.7.2.1', '1.14.13.149', '1.16.3.1', '1.4.3.16', '1.15.1.1', '1.14.12.10', '4.1.1.39', '1.13.11.16', '1.3.3.3', '1.11.1.21'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/159087_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/159087_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1639348.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "12 oxygen-utilizing enzymes were found from classes dict_keys(['1.10.2.2', '1.7.2.1', '1.13.11.49', '1.14.13.208', '1.4.3.16', '1.15.1.1', '4.1.1.39', '1.16.3.1', '1.14.13.2', '1.9.3.1', '1.13.11.8', '1.4.3.5'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1639348_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1639348_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1663591.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "12 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21', '1.14.13.208', '1.9.3.1', '1.14.13.2', '1.7.2.1', '4.1.1.39', '1.10.2.2', '1.16.3.1', '1.13.11.8', '1.4.3.16', '1.4.3.5', '1.15.1.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1663591_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1663591_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1705394.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "5 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.9.3.1', '1.4.3.5', '1.10.2.2', '1.3.3.3'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1705394_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1705394_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1860122.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "5 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21', '1.3.3.3', '1.4.3.16', '4.1.1.39', '1.9.3.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1860122_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1860122_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1904441.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "48 oxygen-utilizing enzymes were found from classes dict_keys(['1.9.3.1', '1.5.3.1', '1.13.11.11', '1.14.99.46', '1.15.1.1', '1.16.3.1', '1.10.3.10', '1.10.2.2', '1.3.3.11', '1.14.13.149', '1.13.11.5', '1.14.12.10', '1.13.11.3', '1.3.99.28', '1.10.3.4', '1.14.13.208', '1.14.13.29', '1.14.13.227', '1.11.1.21', '1.14.14.11', '1.13.11.8', '1.14.12.13', '1.4.3.5', '1.13.11.6', '1.3.3.3', '1.10.3.14', '1.13.12.16', '1.14.13.81', '1.14.12.7', '1.7.2.1', '1.14.13.2', '4.1.1.39', '1.10.3.9', '1.14.11.1', '1.14.14.9', '1.11.1.6', '1.4.3.16', '1.14.13.23', '1.1.3.47', '1.2.3.1', '1.1.3.17', '1.4.3.2', '1.5.3.7', '1.1.3.15', '1.10.3.11', '1.13.11.27', '1.1.5.4', '1.13.11.79'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1904441_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1904441_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1921086.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "3 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21', '1.15.1.1', '4.1.1.39'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1921086_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1921086_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1921087.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "2 oxygen-utilizing enzymes were found from classes dict_keys(['1.15.1.1', '4.1.1.39'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1921087_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/1921087_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/2171755.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "13 oxygen-utilizing enzymes were found from classes dict_keys(['1.15.1.1', '1.9.3.1', '1.14.13.149', '1.5.3.1', '1.14.13.2', '1.13.11.3', '1.11.1.6', '1.3.3.11', '1.7.2.1', '4.1.1.39', '1.10.2.2', '1.4.3.5', '1.13.11.5'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/2171755_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/2171755_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/2177.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "1 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/2177_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/2177_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/2496867.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "4 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.15.1.1', '1.9.3.1', '1.11.1.6'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/2496867_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/2496867_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/259564.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "1 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.6'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/259564_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/259564_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/272942.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "12 oxygen-utilizing enzymes were found from classes dict_keys(['1.3.99.28', '1.15.1.1', '4.1.1.39', '1.4.3.5', '1.11.1.21', '1.10.3.14', '1.9.3.1', '1.13.11.79', '1.14.15.9', '1.16.3.1', '1.5.3.1', '1.10.2.2'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/272942_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/272942_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/272943.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "13 oxygen-utilizing enzymes were found from classes dict_keys(['1.9.3.1', '1.11.1.6', '1.16.3.1', '4.1.1.39', '1.14.14.9', '1.10.2.2', '1.13.11.79', '1.14.13.227', '1.15.1.1', '1.5.3.1', '1.3.3.11', '1.4.3.5', '1.3.99.28'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/272943_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/272943_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/292415.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "8 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.9.3.1', '1.4.3.16', '1.15.1.1', '1.10.2.2', '1.7.2.1', '1.16.3.1', '1.3.3.3'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/292415_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/292415_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/338969.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "14 oxygen-utilizing enzymes were found from classes dict_keys(['1.4.3.5', '1.14.11.33', '1.9.3.1', '1.10.2.2', '1.14.14.5', '1.3.3.3', '1.15.1.1', '1.14.13.2', '1.14.13.82', '1.13.11.8', '1.13.12.16', '1.4.3.16', '1.14.13.208', '4.1.1.39'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/338969_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/338969_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/342108.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "10 oxygen-utilizing enzymes were found from classes dict_keys(['1.15.1.1', '1.14.13.208', '1.16.3.1', '1.4.3.16', '1.4.3.5', '4.1.1.39', '1.9.3.1', '1.7.2.1', '1.10.2.2', '1.14.13.2'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/342108_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/342108_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/349101.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "13 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.6', '1.9.3.1', '1.13.11.79', '1.10.2.2', '1.16.3.1', '1.5.3.1', '1.15.1.1', '4.1.1.39', '1.3.99.28', '1.14.13.227', '1.14.14.9', '1.4.3.5', '1.3.3.11'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/349101_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/349101_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/349102.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "13 oxygen-utilizing enzymes were found from classes dict_keys(['1.7.2.1', '1.11.1.21', '4.1.1.39', '1.9.3.1', '1.16.3.1', '1.10.2.2', '1.3.99.28', '1.5.3.1', '1.15.1.1', '1.4.3.5', '1.13.11.79', '1.11.1.6', '1.3.3.11'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/349102_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/349102_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/365044.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "26 oxygen-utilizing enzymes were found from classes dict_keys(['1.14.13.24', '1.14.13.172', '1.11.1.6', '1.14.13.208', '1.14.12.17', '1.14.12.10', '1.14.12.7', '1.13.11.8', '1.14.14.5', '1.14.13.2', '1.13.11.4', '1.14.12.18', '1.13.11.39', '1.15.1.1', '1.13.11.27', '1.14.13.82', '1.3.3.11', '1.14.14.35', '1.4.3.16', '1.9.3.1', '1.13.11.56', '1.14.12.12', '1.4.3.5', '1.3.3.3', '4.1.1.39', '1.13.11.11'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/365044_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/365044_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/395494.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "8 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21', '1.3.3.3', '1.4.3.16', '1.10.3.14', '1.1.5.4', '4.1.1.39', '1.9.3.1', '1.15.1.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/395494_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/395494_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/395495.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "22 oxygen-utilizing enzymes were found from classes dict_keys(['1.14.13.2', '1.13.11.8', '1.14.13.7', '1.7.2.1', '4.1.1.39', '1.14.13.208', '1.3.3.3', '1.13.11.27', '1.16.3.1', '1.9.3.1', '1.14.13.82', '1.13.11.4', '1.15.1.1', '1.14.13.24', '1.14.13.149', '1.13.11.5', '1.14.13.172', '1.14.16.1', '1.4.3.16', '1.3.3.11', '1.4.3.5', '1.11.1.21'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/395495_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/395495_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/412965.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "4 oxygen-utilizing enzymes were found from classes dict_keys(['1.3.3.3', '4.1.1.39', '1.9.3.1', '1.4.3.5'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/412965_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/412965_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/413404.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "4 oxygen-utilizing enzymes were found from classes dict_keys(['1.3.3.3', '4.1.1.39', '1.9.3.1', '1.4.3.5'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/413404_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/413404_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/431944.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "10 oxygen-utilizing enzymes were found from classes dict_keys(['1.10.2.2', '1.11.1.21', '1.9.3.1', '1.15.1.1', '1.13.11.79', '1.7.2.1', '4.1.1.39', '1.4.3.5', '1.4.3.16', '1.16.3.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/431944_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/431944_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/522306.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "9 oxygen-utilizing enzymes were found from classes dict_keys(['1.4.3.16', '1.4.3.5', '1.7.2.1', '1.11.1.21', '1.15.1.1', '4.1.1.39', '1.9.3.1', '1.3.3.3', '1.10.2.2'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/522306_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/522306_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/547558.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "1 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.21'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/547558_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/547558_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/55518.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "9 oxygen-utilizing enzymes were found from classes dict_keys(['1.16.3.1', '1.9.3.1', '1.4.3.5', '1.11.1.21', '1.15.1.1', '4.1.1.39', '1.4.3.16', '1.7.2.1', '1.10.2.2'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/55518_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/55518_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/57320.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "2 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.16.3.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/57320_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/57320_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/580332.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "8 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.4.3.16', '1.7.2.1', '1.3.3.3', '1.1.5.4', '1.10.3.14', '1.15.1.1', '1.9.3.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/580332_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/580332_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/637389.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "1 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/637389_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/637389_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/643562.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "3 oxygen-utilizing enzymes were found from classes dict_keys(['1.16.3.1', '4.1.1.39', '1.11.1.6'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/643562_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/643562_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/679901.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "2 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.6', '1.15.1.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/679901_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/679901_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/743299.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "3 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.11.1.6', '1.3.3.3'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/743299_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/743299_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/75379.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "10 oxygen-utilizing enzymes were found from classes dict_keys(['4.1.1.39', '1.13.11.24', '1.11.1.21', '1.4.3.5', '1.13.11.11', '1.3.3.3', '1.9.3.1', '1.14.13.149', '1.15.1.1', '1.4.3.16'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/75379_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/75379_EC_CLASSES.txt\n",
      "\n",
      "\n",
      " RUNNING OXYPHEN FOR PROTEOME /Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/990316.fasta\n",
      "Performing Blast searches against oxygen-utilizing database...\n",
      "Filtering Blast output: evalue 0.001  identity 60.0  coverage 60.0\n",
      "\n",
      "\n",
      "\n",
      "2 oxygen-utilizing enzymes were found from classes dict_keys(['1.11.1.6', '1.15.1.1'])\n",
      "Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\n",
      "Executing SVM classifier...\n",
      "\n",
      "\n",
      " OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/990316_oxygen_utilizing_annot.tsv\n",
      "LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN OUTPUT/990316_EC_CLASSES.txt\n",
      "YOUR OUTPUT FILES CAN BE FOUND IN OUTPUT FOLDER\n"
     ]
    }
   ],
   "source": [
    "CONFIG_FILE = open(\"SETTINGS\", \"r\").read().splitlines()\n",
    "\n",
    "if os.path.isfile(\"OUTPUT/results_table.tsv\"):\n",
    "    print (\"OUTPUT FILE EXISTS, PLEASE MOVE/REMOVE YOUR PREVIOUS RESULTS!!!\")\n",
    "    exit()\n",
    "else:\n",
    "    GLOBAL_RESULTS = open(\"OUTPUT/results_table.tsv\", \"a\")\n",
    "\n",
    "def read_config():\n",
    "    multinome_folder=\"\"\n",
    "    for line in CONFIG_FILE:\n",
    "        if line.startswith(\"BLAST_PATH\"):\n",
    "            blast_path = line.split(\"=\")[1]\n",
    "        if line.startswith(\"NUM_THREADS\"):\n",
    "            num_threads = float(line.split(\"=\")[1])\n",
    "        if line.startswith(\"PROTEOMES_FOLDER\") and line.split(\"=\")[1]:\n",
    "            ### enter multinome mode\n",
    "            multinome_folder = line.split(\"=\")[1]\n",
    "\n",
    "    return blast_path, num_threads, multinome_folder\n",
    "\n",
    "def do_oxyphen(proteome, output_filename, ec_classes_file):\n",
    "\n",
    "    '''\n",
    "    Read and parse enzyme.dat file\n",
    "    '''\n",
    "    input_name = \"DATA/enzyme.dat\"\n",
    "    output_name = 'DATA/ec_uniprot.tsv'\n",
    "\n",
    "    ### program ###\n",
    "    handle = open(input_name)\n",
    "    records = Enzyme.parse(handle)\n",
    "\n",
    "    out = dict() #dict of dicts, first key: EC number, second key: field\n",
    "    transferred = dict() #dict of lists\n",
    "    for record in records:\n",
    "        if 'Transferred entry:' in record['DE']:\n",
    "            record['DE'] = record['DE'].rstrip('.') #remove period\n",
    "            record['DE'] = record['DE'].replace('Transferred entry:',' ') #remove title\n",
    "            record['DE'] = record['DE'].replace(',',' ') #remove commas\n",
    "            record['DE'] = record['DE'].replace('and',' ') #remove and\n",
    "            point_to = record['DE'].split()\n",
    "            transferred[record['ID']] = point_to\n",
    "        else:\n",
    "            out[record['ID']] = dict()\n",
    "            out[record['ID']]['uniprot'] = ' '.join([x[0] for x in record['DR']])\n",
    "            out[record['ID']]['description'] = record['DE']\n",
    "            out[record['ID']]['transferred'] = False\n",
    "\n",
    "    for id in transferred:\n",
    "        out[id] = dict()\n",
    "        out[id]['uniprot'] = ' '.join([out[x]['uniprot'] for x in transferred[id]])\n",
    "        out[id]['description'] = 'Transferred entry: ' + ' '.join(transferred[id])\n",
    "        out[id]['transferred'] = True\n",
    "    df = pd.DataFrame.from_dict(out, orient='index')\n",
    "    df.index.name = 'EC'\n",
    "    df.to_csv(output_name, sep='\\t')\n",
    "    \n",
    "\n",
    "    '''\n",
    "    Take a subset of ecs of interest\n",
    "    '''\n",
    "\n",
    "    oxidases = tuple(open(\"DATA/oxygen_ecclasses\", \"r\").read().splitlines())\n",
    "\n",
    "    infile = open(\"DATA/ec_uniprot.tsv\", \"r\").readlines()\n",
    "    outfile = open(\"DATA/ec_uniprot_oxidases.tsv\", \"w\")\n",
    "\n",
    "    for line in infile:\n",
    "        if line.startswith(\"EC\"):\n",
    "            outfile.write(line)\n",
    "        elif line.startswith(oxidases):\n",
    "            outfile.write(line)\n",
    "\n",
    "    outfile.close()\n",
    "    \n",
    "\n",
    "    '''\n",
    "    write a file with one uniprot ID per line, containing all of the\n",
    "    uniprot IDs mentioned in uniprot column of the input file\n",
    "\n",
    "    Ignore EC numbers that have been transferred\n",
    "    '''\n",
    "\n",
    "    input = \"DATA/ec_uniprot_oxidases.tsv\"\n",
    "    output = \"DATA/uniprot_ids.txt\"\n",
    "\n",
    "    df = pd.read_table(input)\n",
    "    df.dropna(subset=['uniprot'], inplace=True) #ignore EC numbers with no uniprot ids associated\n",
    "\n",
    "    df = df[df.transferred == False] #ignore EC numbers that are obsolete due to transfer\n",
    "\n",
    "    unique_uniprot = set(\" \".join(df.uniprot.values).split(\" \"))\n",
    "\n",
    "    with open(output, \"w\") as outfile:\n",
    "        for id in unique_uniprot:\n",
    "            outfile.write(id + \"\\n\")\n",
    "    outfile.close()\n",
    "    \n",
    "\n",
    "    '''\n",
    "    Make blastdb out of the swissprot subset\n",
    "    '''\n",
    "\n",
    "    blast_path, num_threads, multinome_folder = read_config()\n",
    "\n",
    "    os.system(\"%s -in DATA/sprot_subset.fasta -dbtype prot -out DATA/sprot_subset -hash_index\" % (os.path.join(blast_path, \"makeblastdb\")))\n",
    "\n",
    "    '''\n",
    "    Blast our pre-selected proteomes against the uniprot subset\n",
    "    '''\n",
    "    #proteome = \"/Users/tonglen/Downloads/oxyphen-multiome/PROTEOMES/1131.fasta\"\n",
    "    print (\"Performing Blast searches against oxygen-utilizing database...\")\n",
    "    os.system(\"%s -max_target_seqs 1 -outfmt '6 qseqid sseqid pident evalue qcovs' -query %s -db DATA/sprot_subset -out DATA/new_sequences_sprot_enzyme.tab -num_threads %d\" % (os.path.join(blast_path, \"blastp\"), proteome, num_threads) )\n",
    "   #os.system(\"%s -max_target_seqs 1 -outfmt '6 qseqid sseqid pident evalue qcovs' -query %s -db DATA/sprot_subset -out DATA/new_sequences_sprot_enzyme.tab -num_threads %d\" % (os.path.join(blast_path, \"blastp\"), proteome, num_threads) )\n",
    "\n",
    "\n",
    "    '''\n",
    "    Filter Blast output.\n",
    "    '''\n",
    "    evalue = 0.001\n",
    "    identity = 60.0\n",
    "    coverage = 60.0\n",
    "\n",
    "    print (\"Filtering Blast output: evalue\",evalue, \" identity\", identity, \" coverage\", coverage)\n",
    "    hits_table_file_name = \"DATA/new_sequences_sprot_enzyme.tab\"\n",
    "    hits_table_file_name_filtered_out = open(\"DATA/new_sequences_sprot_enzyme_filtered.tab\", \"w\")\n",
    "\n",
    "    hits_table_file_name_filtered_out.write(\"\\t\".join([\"hit\",\"subject\",\"id\",\"len\",\"eval\",\"cov\"])+\"\\n\")\n",
    "\n",
    "    for line in open(hits_table_file_name, \"r\").read().splitlines():\n",
    "        if line.startswith(\"#\"): continue\n",
    "\n",
    "        query, target, ident, eval, cover = line.split(\"\\t\")\n",
    "        eval = float(eval)\n",
    "        ident = float(ident)\n",
    "        cover = float(cover)\n",
    "\n",
    "        if eval <= evalue and ident >= identity and cover >= coverage:\n",
    "            hits_table_file_name_filtered_out.write(line+\"\\n\")\n",
    "\n",
    "    hits_table_file_name_filtered_out.close()\n",
    "    \n",
    "    \n",
    "\n",
    "    hits_table_file_name_filtered = \"DATA/new_sequences_sprot_enzyme_filtered.tab\"\n",
    "    enzyme_table_file_name = 'DATA/ec_uniprot_oxidases.tsv'\n",
    "\n",
    "    hits = pd.read_csv(hits_table_file_name_filtered, sep=\"\\t\", header=0)\n",
    "    enzyme = pd.read_csv(enzyme_table_file_name, sep=\"\\t\", header=0)\n",
    "\n",
    "    hits.fillna('', inplace=True)  #replace empty values with blank spaces\n",
    "    enzyme.fillna('', inplace=True)\n",
    "\n",
    "    enzyme = enzyme[enzyme.transferred == False] #drop transferred EC numbers\n",
    "\n",
    "    hits.subject = hits.subject.str[3:9] #take just the uniprot ID from the name\n",
    "\n",
    "    def get_ecs(uniprot):\n",
    "        if uniprot == '': #ignore invalid uniprot ids\n",
    "            return ''\n",
    "        else:\n",
    "            return ' '.join(enzyme.EC[enzyme.uniprot.str.contains(uniprot)].values)\n",
    "\n",
    "    hits['EC'] = hits.subject.apply(get_ecs)\n",
    "\n",
    "    output_file_name = output_filename\n",
    "    hits.to_csv(output_file_name, sep=\"\\t\", index=False)\n",
    "    \n",
    "\n",
    "\n",
    "    ### read final mapping output\n",
    "\n",
    "    mapping_out = open(output_file_name, \"r\").read().splitlines()\n",
    "    ecs_dict = {}\n",
    "\n",
    "    for line in mapping_out[1:]:\n",
    "        splitted = line.split(\"\\t\")\n",
    "        ecs = splitted[-1]\n",
    "\n",
    "        for ec in ecs.split():\n",
    "            if ec not in ecs_dict:\n",
    "                ecs_dict[ec] = []\n",
    "            ecs_dict[ec].append(splitted[0])\n",
    "\n",
    "    print (\"\\n\\n\")\n",
    "    print (len(ecs_dict), \"oxygen-utilizing enzymes were found from classes\", ecs_dict.keys())\n",
    "    \n",
    "    ec_out = open(ec_classes_file, \"w\")  \n",
    "    ec_out.write(\"\\t\".join(ecs_dict.keys()))\n",
    "\n",
    "    ec_out.close()\n",
    "    \n",
    "    GLOBAL_RESULTS.write(os.path.basename(proteome)+\"\\t\"+str(len(ecs_dict))+\"\\t\"+\",\".join(ecs_dict.keys())+\"\\n\")\n",
    "    \n",
    "    print (\"Detailed mapping can be found in OUTPUT/oxygen_utilizing_annot.tsv file\")\n",
    "    print (\"Executing SVM classifier...\")\n",
    "\n",
    "    infile = open(\"DATA/model_svm\", \"r\").read().splitlines()\n",
    "\n",
    "    classifier_input = []\n",
    "    classes = []\n",
    "    ec_classes = []\n",
    "\n",
    "    for line in infile:\n",
    "\n",
    "        if line.startswith(\"@attribute\") and \"class\" not in line:\n",
    "            ec_classes.append(line.split()[1].replace(\"'\",\"\"))\n",
    "\n",
    "'''\n",
    "Execute oxyphen for all proteomes in the input folder\n",
    "'''            \n",
    "\n",
    "def do_for_all():\n",
    "    blast_path, num_threads, multinome_folder, = read_config()  \n",
    "    proteomes = glob.glob(os.path.join(multinome_folder, \"*\"))\n",
    "    #proteomes = glob.glob(os.path.join(\"Users/tonglen/Downloads/oxyphen-multiome/TEST\", \"*.fasta\")\n",
    "    #print (\"PROTEOMES IN YOUR PROTEOMES_FOLDER DIRECTORY:\")\n",
    "\n",
    "    for proteome in proteomes:\n",
    "        fname = os.path.splitext(os.path.basename(proteome))[0]\n",
    "        output_filename = os.path.join(\"OUTPUT\",fname+\"_oxygen_utilizing_annot.tsv\")\n",
    "\n",
    "        ec_classes_file = os.path.join(\"OUTPUT\",fname+\"_EC_CLASSES.txt\")\n",
    "\n",
    "        print (\"\\n\\n RUNNING OXYPHEN FOR PROTEOME %s\" % proteome)\n",
    "        do_oxyphen(proteome, output_filename, ec_classes_file)\n",
    "\n",
    "        print (\"\\n\\n OUTPUT MAPPING FILE FOR THIS PROTEOME CAN BE FOUND IN %s\" % output_filename)\n",
    "        print (\"LIST OF EC CLASSES FOR THIS PROTEOME CAN BE FOUND IN %s\" % ec_classes_file)\n",
    "        \n",
    "    GLOBAL_RESULTS.close()\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    do_for_all()\n",
    "\n",
    "print (\"YOUR OUTPUT FILES CAN BE FOUND IN OUTPUT FOLDER\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
